doc/source/development/contributing_codebase.rst

```diff
@@ -402,10 +402,12 @@ pandas uses `mypy <http://mypy-lang.org>`_ and `pyright <https://github.com/micr
    mypy pandas
 
    # let pre-commit setup and run pyright
-   pre-commit run --all-files pyright
+   pre-commit run --hook-stage manual --all-files pyright
    # or if pyright is installed (requires node.js)
    pyright
 
+A recent version of ``numpy`` (>=1.21.0) is required for type validation.
+
```

Comment:
Does the `pyright@1.1.171` dependency include this? Because if not, as it's running as a local hook, then it won't have access to `numpy`, even if you have it installed in your virtual environment

Perhaps it should be `repo: system`?

(granted, I haven't tried running this, am just pointing out a common misconception about `repo: local`)

Reply:
It seems that pre-commit does not isolate pyright from the current python environment: if numpy is installed in the currently active environment, pyright seems to have access to it.

I tried adding it as a dependency (in the pre-commit), but it seems that pre-commit does not support mixing languages (pyright: node, numpy:python) - it tries to use node to install numpy.

Reply:
> It seems that pre-commit does not isolate pyright from the current python environment: if numpy is installed in the currently active environment, pyright seems to have access to it.

I find this surprising, but if that's the case, then great 👍